input/listen.go

```diff
@@ -1,40 +1,62 @@
 package input
 
-import "net"
+import (
+	"bytes"
+	"net"
+)
 
-func listen(addr string, handler Handler) (net.Listener, error) {
+func listen(addr string, handler Handler) error {
 	laddr, err := net.ResolveTCPAddr("tcp", addr)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	l, err := net.ListenTCP("tcp", laddr)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	log.Notice("listening on %v/tcp", laddr)
-	go accept(l, handler)
+	go acceptTcp(l, handler)
 
 	udp_addr, err := net.ResolveUDPAddr("udp", addr)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	udp_conn, err := net.ListenUDP("udp", udp_addr)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	log.Notice("listening on %v/udp", udp_addr)
-	go handler.Handle(udp_conn)
+	go acceptUdp(udp_conn, handler)
 
-	return l, nil
+	return nil
 }
 
-func accept(l *net.TCPListener, handler Handler) {
+func acceptTcp(l *net.TCPListener, handler Handler) {
 	for {
 		c, err := l.AcceptTCP()
 		if err != nil {
 			log.Error(err.Error())
 			break
 		}
-		go handler.Handle(c)
+		go acceptTcpConn(c, handler)
+	}
+}
+
+func acceptTcpConn(c net.Conn, handler Handler) {
+	defer c.Close()
+	log.Debug("listen.go: tcp connection from %v", c.RemoteAddr())
+	handler.Handle(c)
+}
+
+func acceptUdp(l *net.UDPConn, handler Handler) {
+	for {
+		buffer := make([]byte, 4096)
```

Comment:
What happens if clients send payloads that are larger then 4096 bytes? Though they shouldnt, they can send up to 65k bytes.  Will everything after 4096 just be dropped? or will the l.ReadFrom() continue reading the UDP datagram on the next loop iteration.  If the data is dropped, we need to document that limitation.  If we continue reading the datagram on the next loop iteration, then we will end up with bad data.  eg if the packet is split at `...some.m] - [etric.foo 1 12345678` then `some.m` would be dropped by the handler and `etric.foo` would be processed creating a new series that shouldnt be there.

Reply:
Good question, I'll do some testing here with large udp packets